crypto/ec/curve25519.c

```diff
@@ -4425,1066 +4521,1086 @@ static const ge_precomp Bi[8] = {
+    int64_t a4  = 0x1fffff & (load_4(a + 10) >> 4);
+    int64_t a5  = 0x1fffff & (load_3(a + 13) >> 1);
+    int64_t a6  = 0x1fffff & (load_4(a + 15) >> 6);
+    int64_t a7  = 0x1fffff & (load_3(a + 18) >> 3);
+    int64_t a8  = 0x1fffff &  load_3(a + 21);
+    int64_t a9  = 0x1fffff & (load_4(a + 23) >> 5);
+    int64_t a10 = 0x1fffff & (load_3(a + 26) >> 2);
+    int64_t a11 =            (load_4(a + 28) >> 7);
+    int64_t b0  = 0x1fffff &  load_3(b);
+    int64_t b1  = 0x1fffff & (load_4(b +  2) >> 5);
+    int64_t b2  = 0x1fffff & (load_3(b +  5) >> 2);
+    int64_t b3  = 0x1fffff & (load_4(b +  7) >> 7);
+    int64_t b4  = 0x1fffff & (load_4(b + 10) >> 4);
+    int64_t b5  = 0x1fffff & (load_3(b + 13) >> 1);
+    int64_t b6  = 0x1fffff & (load_4(b + 15) >> 6);
+    int64_t b7  = 0x1fffff & (load_3(b + 18) >> 3);
+    int64_t b8  = 0x1fffff &  load_3(b + 21);
+    int64_t b9  = 0x1fffff & (load_4(b + 23) >> 5);
+    int64_t b10 = 0x1fffff & (load_3(b + 26) >> 2);
+    int64_t b11 =            (load_4(b + 28) >> 7);
+    int64_t c0  = 0x1fffff &  load_3(c);
+    int64_t c1  = 0x1fffff & (load_4(c +  2) >> 5);
+    int64_t c2  = 0x1fffff & (load_3(c +  5) >> 2);
+    int64_t c3  = 0x1fffff & (load_4(c +  7) >> 7);
+    int64_t c4  = 0x1fffff & (load_4(c + 10) >> 4);
+    int64_t c5  = 0x1fffff & (load_3(c + 13) >> 1);
+    int64_t c6  = 0x1fffff & (load_4(c + 15) >> 6);
+    int64_t c7  = 0x1fffff & (load_3(c + 18) >> 3);
+    int64_t c8  = 0x1fffff &  load_3(c + 21);
+    int64_t c9  = 0x1fffff & (load_4(c + 23) >> 5);
+    int64_t c10 = 0x1fffff & (load_3(c + 26) >> 2);
+    int64_t c11 =            (load_4(c + 28) >> 7);
+    int64_t s0;
+    int64_t s1;
+    int64_t s2;
+    int64_t s3;
+    int64_t s4;
+    int64_t s5;
+    int64_t s6;
+    int64_t s7;
+    int64_t s8;
+    int64_t s9;
+    int64_t s10;
+    int64_t s11;
+    int64_t s12;
+    int64_t s13;
+    int64_t s14;
+    int64_t s15;
+    int64_t s16;
+    int64_t s17;
+    int64_t s18;
+    int64_t s19;
+    int64_t s20;
+    int64_t s21;
+    int64_t s22;
+    int64_t s23;
+    int64_t carry0;
+    int64_t carry1;
+    int64_t carry2;
+    int64_t carry3;
+    int64_t carry4;
+    int64_t carry5;
+    int64_t carry6;
+    int64_t carry7;
+    int64_t carry8;
+    int64_t carry9;
+    int64_t carry10;
+    int64_t carry11;
+    int64_t carry12;
+    int64_t carry13;
+    int64_t carry14;
+    int64_t carry15;
+    int64_t carry16;
+    int64_t carry17;
+    int64_t carry18;
+    int64_t carry19;
+    int64_t carry20;
+    int64_t carry21;
+    int64_t carry22;
+
+    s0 = c0 + a0 * b0;
+    s1 = c1 + a0 * b1 + a1 * b0;
+    s2 = c2 + a0 * b2 + a1 * b1 + a2 * b0;
+    s3 = c3 + a0 * b3 + a1 * b2 + a2 * b1 + a3 * b0;
+    s4 = c4 + a0 * b4 + a1 * b3 + a2 * b2 + a3 * b1 + a4 * b0;
+    s5 = c5 + a0 * b5 + a1 * b4 + a2 * b3 + a3 * b2 + a4 * b1 + a5 * b0;
+    s6 = c6 + a0 * b6 + a1 * b5 + a2 * b4 + a3 * b3 + a4 * b2 + a5 * b1 +
+        a6 * b0;
```

Comment:
Shouldn't `a6` be lined up with `c6`? Similar comment for lines below.

Reply:
Hmm... in this hunk I didn't do any aligning (yet), I only shortened overly long lines, which went from 80 to 82 bytes by the reindentation. I kept the original indation of 4 bytes for followup lines. You can see this nicely in https://github.com/openssl/openssl/commit/9c8ffa66cae4aa4c519c1344f73d0f2a9ff6c858?diff=split.

I could align the followup lines, but then I would have to do this for the entire block. Also note that there are different types of lines, the ones starting with a carry

https://github.com/openssl/openssl/blob/9c8ffa66cae4aa4c519c1344f73d0f2a9ff6c858/crypto/ec/curve25519.c#L4989-L4992

and the ones without:

https://github.com/openssl/openssl/blob/9c8ffa66cae4aa4c519c1344f73d0f2a9ff6c858/crypto/ec/curve25519.c#L5005-L5008

If I would indent 5 bytes instead of 4, the `*` operators would align nicely in the latter case, but in the former the `*` operators would align with `+` operators which does not make sense. That's why I left it unchanged. But If you think it is better that way, or if you have an alternative proposal, I'm open for suggestions.

Reply:
> > I fear that perfect alignment of variables competes with our requiring 4 space indent of the broken lines.
> 
> Is that a requirement??

@levitte @mattcaswell hang on for a minute, I think I found a nice compromise...